platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/transients/field/WorkflowUserField.java

```diff
@@ -61,22 +63,65 @@ private List<String> getUserTaskAgeStrings(R row) {
 			.collect(Collectors.toList());
 	}
 
-	private Map<AGUser, DayTime> getUserTaskCreationMap(Set<AGWorkflowTask> tasks) {
+	private Map<UserAndDelegationUser, DayTime> getUserTaskCreationMap(Set<AGWorkflowTask> tasks) {
 
 		AGWorkflowTask.CREATION_TRANSACTION.prefetchAll(tasks);
 
-		TreeMap<AGUser, DayTime> userTaskCreationMap = new TreeMap<>();
+		Map<UserAndDelegationUser, DayTime> userTaskCreationMap = new HashMap<>();
 		for (AGWorkflowTask task: tasks) {
 			Optional<AGTransaction> creationTransaction = Optional.ofNullable(AGWorkflowTask.CREATION_TRANSACTION.getValue(task));
-			if (creationTransaction.isPresent()) {
-				userTaskCreationMap.merge(task.getUser(), creationTransaction.get().getAt(), DayTime::min);
+
+			if (creationTransaction.isPresent() && !task.hasActiveDelegation()) {
+				userTaskCreationMap.merge(new UserAndDelegationUser(task.getUser(), task.getUser()), creationTransaction.get().getAt(), DayTime::min);
+			} else if (task.hasActiveDelegation()) {
+				AGWorkflowTaskDelegation delegation = task.getActiveDelegation().get();
+				Optional<DayTime> lastModificationTimeFormLog = delegation.getLastModificationTimeFormLog();
+				if (lastModificationTimeFormLog.isPresent()) {
+					userTaskCreationMap
+						.merge(new UserAndDelegationUser(task.getUser(), delegation.getTargetUser()), lastModificationTimeFormLog.get(), DayTime::min);
+				}
 			}
+
 		}
 		return userTaskCreationMap;
 	}
 
-	private String getUserTaskAgeString(AGUser user, int days) {
+	private String getUserTaskAgeString(UserAndDelegationUser userAndCurrentDelegation, int days) {
+
+		if (userAndCurrentDelegation.getUserDefinedInTask() == userAndCurrentDelegation.getUserInCharge()) {
+			return String
+				.format(
+					"%s (%s %s)",
+					userAndCurrentDelegation.getUserDefinedInTask().toDisplayWithoutId(),
+					days,
+					days == 1? WorkflowI18n.DAY : WorkflowI18n.DAYS);
+		} else {
+			return String
+				.format(
+					"%s -> %s (%s %s)",
+					userAndCurrentDelegation.getUserDefinedInTask().toDisplayWithoutId(),
+					userAndCurrentDelegation.getUserInCharge().toDisplayWithoutId(),
+					days,
+					days == 1? WorkflowI18n.DAY : WorkflowI18n.DAYS);
+		}
+
+	}
+
+	private static class UserAndDelegationUser extends Pair<AGUser, AGUser> {
```

Comment:
- Since this class is private anyway:
  - I suggest to rename it to `Delegation`, for the sake of simplicity.
  - There's little point in reducing the visibilities of CTOR and methods.
- We should care less about where the data came from, and more about the semantics of the delegation as such. Therefore suggesting the following renaming:
  - `userDefinedInTask` -> `originalUser` **or** `delegator`
  - `userInCharge` -> `delegatee`

Reply:
got suggestions => changed